scripts/pkglib.sh

```diff
@@ -164,6 +164,43 @@ function checkout_repo()
     cp ../scripts/apport_hook.py $OBS_WORKDIR/$APPORT_HOOK
 }
 
+function generate_kernel_config() {
+    local kernel_version=$1
+    [[ ! "$kernel_version" =~ ^[0-9]\.[0-9]{1,3}\.[0-9]{1,3}$ ]] && die "Wrong kernel version supplied"
+
+    local kernel_filename=$2
+    [ -z "$kernel_filename" ] && die "No kernel config filename supplied"
+    [ -e "$kernel_filename" ] && echo "INFO: ${kernel_filename} exists, it will be overwritten!"
+
+    # Temporary directory to decompress the kernel
+    local tmpdir=$(mktemp -d)
+
+    local kernel_tarball_suffix=".tar.xz"
+    local kernel_url=https://www.kernel.org/pub/linux/kernel/v"${KR_SERIES}"/linux-"${kernel_version}${kernel_tarball_suffix}"
+    local kernel_tree_dir="${tmpdir}/linux-${kernel_version}"
+
+
+    # Download the kernel source tree tarball and decompress it in ${tmpdir}
+    [ ! -e "${kernel_tree_dir}${kernel_tarball_suffix}" ] &&
+        curl -o "${kernel_tree_dir}${kernel_tarball_suffix}" -L $kernel_url || true
+    tar -C $tmpdir -xf "${kernel_tree_dir}${kernel_tarball_suffix}"
+
+    # Generate a new kernel config file based on the one in this repo
+    cp kernel-config-${KR_LTS}* "${kernel_tree_dir}/.config"
+    pushd "${kernel_tree_dir}"
+    make -s ARCH=x86_64 oldconfig
+    popd
+
+    # Remove old config and replace it with the new one
+    rm -f ./kernel-config-${KR_LTS}*
+    cp "${kernel_tree_dir}/.config" $kernel_filename
+    git add "$kernel_filename"
+    git commit -s -m "$(echo -e "kernel: automatic config update\\n\\nApplied 'make oldconfig' changes.")" "$kernel_filename"
+
+    # remove the kernel tree
+    rm -rf "${kernel_tree_dir}/"
```

Comment:
can you remove the `/` just in case for some reason be empty not try to remove the root directory